packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -0,0 +1,562 @@
+    width,
+    height
+  } = props;
+
+  let {addGlobalListener, removeGlobalListener} = useGlobalListeners();
+
+  let {direction, locale} = useLocale();
+
+  let colorAreaDimensions = {width, height};
+
+  let focusedInputRef = useRef<HTMLElement>(null);
+
+  let focusInput = useCallback((inputRef = inputXRef) => {
+    if (inputRef.current) {
+      focusWithoutScrolling(inputRef.current);
+    }
+  }, [inputXRef]);
+
+  let stateRef = useRef<ColorAreaState>(null);
+  stateRef.current = state;
+  let channels = stateRef.current.getChannels();
+  if (!xChannel || !yChannel) {
+    xChannel = channels.xChannel;
+    yChannel = channels.yChannel;
+  }
+  let zChannel = channels.zChannel;
+
+  let currentPosition = useRef<{x: number, y: number}>(null);
+
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      if (!e.shiftKey && /^Arrow(?:Right|Left|Up|Down)$/.test(e.key)) {
+        return;
+      }
+      let stepSize = Math.max(step, CHANNEL_STEP_SIZE[xChannel]);
+      let range = stateRef.current.value.getChannelRange(xChannel);
+      switch (e.key) {
+        case 'PageUp':
+        case 'ArrowUp':
+          range = stateRef.current.value.getChannelRange(yChannel);
+          stepSize = Math.max(step, CHANNEL_STEP_SIZE[yChannel]);
+          stateRef.current.setYValue(
+            clamp(
+              (Math.floor(stateRef.current.yValue / stepSize) + 1) * stepSize,
+              range.minValue,
+              range.maxValue
+            )
+          );
+          focusedInputRef.current = inputYRef.current;
+          break;
+        case 'PageDown':
+        case 'ArrowDown':
+          range = stateRef.current.value.getChannelRange(yChannel);
+          stepSize = Math.max(step, CHANNEL_STEP_SIZE[yChannel]);
+          stateRef.current.setYValue(
+            clamp(
+              (Math.ceil(stateRef.current.yValue / stepSize) - 1) * stepSize,
+              range.minValue,
+              range.maxValue
+            )
+          );
+          focusedInputRef.current = inputYRef.current;
+          break;
+        case 'Home':
+        case 'ArrowLeft':
+          stateRef.current.setXValue(
+            clamp(
+              (Math[direction === 'rtl' ? 'floor' : 'ceil'](stateRef.current.xValue / stepSize) + (direction === 'rtl' ? 1 : -1)) * stepSize,
+              range.minValue,
+              range.maxValue
+            )
+          );
+          focusedInputRef.current = inputXRef.current;
+          break;
+        case 'End':
+        case 'ArrowRight':
+          stateRef.current.setXValue(
+            clamp(
+              (Math[direction === 'rtl' ? 'floor' : 'ceil'](stateRef.current.xValue / stepSize) + (direction === 'rtl' ? -1 : 1)) * stepSize,
+              range.minValue,
+              range.maxValue
+            )
+          );
+          focusedInputRef.current = inputXRef.current;
+          break;
+      }
+      if (focusedInputRef.current) {
+        e.preventDefault();
+        setTimeout(() => {
```

Comment:
Had some strangeness using VoiceOver with Home/End keys, because ArrowRight with `fn` key down would cause VoiceOver to focus the next input element in the DOM rather than keeping focus on the horizontal slider. End key would end up getting announced as if the yChannel value had changed rather than the xChannel value.